pii-manager/src/pii_manager/api/file.py

```diff
@@ -0,0 +1,133 @@
+"""
+File-based API
+"""
+
+import sys
+import re
+import json
+import gzip
+import bz2
+import lzma
+from itertools import zip_longest
+
+from typing import Dict, List, TextIO, Iterable, Optional, Union
+
+from pii_manager.api import PiiManager
+from pii_manager.helper.exception import PiiManagerException
+
+
+def openfile(name: str, mode: str) -> TextIO:
+    """
+    Open files, raw text or compressed (gzip, bzip2 or xz)
+    """
+    name = str(name)
+    if name == "-":
+        return sys.stdout if mode.startswith("w") else sys.stdin
+    elif name.endswith(".gz"):
+        return gzip.open(name, mode, encoding="utf-8")
+    elif name.endswith(".bz2"):
+        return bz2.open(name, mode, encoding="utf-8")
+    elif name.endswith(".xz"):
+        return lzma.open(name, mode, encoding="utf-8")
+    else:
+        return open(name, mode, encoding="utf-8")
+
+
+def sentence_splitter(doc: str) -> Iterable[str]:
+    """
+    Split text by sentence separators
+     (keeping the separator at the end of the sentence, so that joining the
+    pieces recovers exactly the same text)
+    """
+    split = re.split(r"(\s*[\.!\?．。]\s+)", doc)
```

Comment:
This might not work as expected for some languages and/or non-standard text (ie chats, forums or generally user generated content where punctuation is not strictly used.).

If possible sentence splitting should be an external function that could be adapted per language or source.

Reply:
that's right, split-by-sentence can get tricky. For this reason there are two other split modes: `line` and `block`. Block is the safest, since it processes the full document, but if the document is rather big, it can be a memory hog. Line is easiest, but of course if a PII is split across lines, it won't detect it.

having a pluggable splitter could indeed be a good idea, Note though that this is for the file-based API; in the object-based API you supply chunks of text on your own, so you can split externally with whatever procedure you see fit, and pass the split chunks to the package